src/language-html/parser-html.js

```diff
@@ -339,7 +354,7 @@ function createParser({
   normalizeAttributeName = false,
   allowHtmComponentClosingTags = false,
   isTagNameCaseSensitive = false,
-  getTagContentType,
+  getTagContentType = null,
```

Comment:
I don't think this is right, default value of `getTagContentType` should be `undefined`, `null` don't make sense.

Reply:
Done, hope it doesn't break my build

Reply:
Results in this error now:

```
  ✘  unicorn/no-useless-undefined

     Do not use useless `undefined`
```

I am looking into some alternatives now.

Reply:
I just pushed a more radical soution:

```diff
--- a/src/language-html/parser-html.js
+++ b/src/language-html/parser-html.js
@@ -348,24 +348,9 @@ function locEnd(node) {
   return node.sourceSpan.end.offset;
 }
 
-function createParser({
-  recognizeSelfClosing = false,
-  normalizeTagName = false,
-  normalizeAttributeName = false,
-  allowHtmComponentClosingTags = false,
-  isTagNameCaseSensitive = false,
-  getTagContentType,
-} = {}) {
+function createParser(parserOptions = {}) {
   return {
-    parse: (text, parsers, options) =>
-      _parse(text, options, {
-        recognizeSelfClosing,
-        normalizeTagName,
-        normalizeAttributeName,
-        allowHtmComponentClosingTags,
-        isTagNameCaseSensitive,
-        getTagContentType,
-      }),
+    parse: (text, parsers, options) => _parse(text, options, parserOptions),
     hasPragma,
     astFormat: "html",
     locStart,
```

Reply:
Current one  is  absolutely not  right , we  may  pass  extra option to  createParser.

This  type check ,to me , looks like  saying `cont  {foo} = {}` , foo  must  have a default value , isn't `foo` already a  `undefined`?

Reply:
I am not sure if I am understanding your comments correctly. The primary purpose of the simplification is to avoid issues between JSDoc typecheck & existing eslint rules. I would also favor not keeping another internal options interface layer if it is not really needed.

Does the proposed simplification look OK, or are there any issues with it?

Reply:
1. `createParser({getTagContentType = null,})` This could break  `angular-html-parser`, if it do `foo({getTagContentType = somethingNotNull})`

2. `createParser({getTagContentType = undefined,})` this `undefined` seems useless

3. `createParser(parserOptions)` is not right, it's not `parserOptions`, it's could be anything, for example we could pass `prepocess` function

Reply:
I just backed these changes out and added a couple more `@ts-ignore` comments. I think it would be nice to improve it someday later.